osf/models/files.py

```diff
@@ -445,7 +445,7 @@ def update(self, revision, data, user=None):
                 break
         else:
             # Insert into history if there is no matching etag
-            if data.get('modified'):
+            if data.get('modified') and None not in [item['modified'] for item in self.history]:
```

Comment:
After chatting with @felliott, we're going to modify the behavior so that if you have etags, great, if not, then just add it to the end of the list. However, Fitz has a request for a check that will bubble up a sentry error so we can see if we ever hit a particular state. Once he's added the description of the check, we can move this to addl dev.